src/soundfile.js

```diff
@@ -539,6 +539,8 @@ define(function (require) {
     var pTime = time + now;
 
     if (this.isPlaying() && this.buffer && this.bufferSourceNode) {
+      this._workletNode.port.postMessage({ name: 'pause' });
+
       this.pauseTime = this.currentTime();
       this.bufferSourceNode.stop(pTime);
       this._counterNode.stop(pTime);
```

Comment:
one issue we should be aware of is that a pause can be scheduled in the future...perhaps a reason to make `pause` an a-rate param so that we could setValueAtTime?

Reply:
if we move the pause logic out of the worklet processor and into the SoundFile, is that still an issue?

Reply:
Yes, unfortunately that wouldn’t solve the issue. It was an issue before this pr. Just thinking whether keeping pause logic in the processor could help fix it.

Reply:
I tried making `pause` an a-rate AudioParam that would get converted to a boolean in the worklet processor, but I had trouble getting that approach to work properly, even in Chrome. For some reason, `setValueAtTime` wasn't changing the value of the AudioParam.

One way to handle this problem with the approach that I settled on (see the comment thread above) could be to modify this [guard statement](https://github.com/processing/p5.js-sound/commit/ce48b1b4c12241fc2ce3445331fe9a06c502f872#diff-635fd3ddb6f066be99025f1994db3e30R1251) to only check `event.position.data === 0` rather than `this._paused || event.position.data === 0`.

`event.data.position` should only equal 0 when the counter buffer has actually stopped playing, which would be compatible with scheduling a pause in the future. I'm not 100% sure this would work, but it's my best guess...

Reply:
nice solution! At some point, I'd be curious to experiment more with `setValueAtTime` on worklet AudioParams, but this works for now.

Either way, it seems like we only need to check `event.position.data === 0` in the guard statement?

Reply:
I believe so, yes. I changed the guard statement to only check `event.position.data === 0` and it seems to work on Chrome, Firefox, and Safari.